Headers/AppLovinSDK.framework/Headers/ALSdk.h

```diff
@@ -0,0 +1,226 @@
+ */
+- (void)showMediationDebugger;
+
+#pragma mark - SDK Initialization
+
+/**
+ * The sort of callback that the SDK calls when it finishes initializing.
+ */
+typedef void (^ALSdkInitializationCompletionHandler)(ALSdkConfiguration *configuration);
+
+/**
+ * Initializes the SDK.
+ */
+- (void)initializeSdk;
+
+/**
+ * Initializes the SDK with a given completion block.
+ *
+ * The SDK invokes the callback on the main thread.
+ *
+ * @param completionHandler The callback that the SDK will call when the SDK finishes initializing.
+ *
+ * @see <a href="https://dash.applovin.com/documentation/mediation/ios/getting-started/integration#initialize-the-sdk">MAX Integration Guide ⇒ iOS ⇒ Integration ⇒ Initialize the SDK</a>
+ */
+- (void)initializeSdkWithCompletionHandler:(nullable ALSdkInitializationCompletionHandler)completionHandler;
+
+/**
+ * Initializes the default instance of AppLovin SDK.
+ *
+ * @warning Make sure your SDK key is set in the application’s `Info.plist` under the property `AppLovinSdkKey`.
+ *
+ * @see <a href="https://dash.applovin.com/documentation/mediation/ios/getting-started/integration#initialize-the-sdk">MAX Integration Guide ⇒ iOS ⇒ Integration ⇒ Initialize the SDK</a>
+ */
++ (void)initializeSdk;
+
+/**
+ * Initializes the default instance of AppLovin SDK.
+ *
+ * @warning Make sure your SDK key is set in the application’s `Info.plist` under the property `AppLovinSdkKey`.
+ *
+ * @param completionHandler The callback that the SDK will run on the main queue when the SDK finishes initializing.
+ *
+ * @see <a href="https://dash.applovin.com/documentation/mediation/ios/getting-started/integration#initialize-the-sdk">MAX Integration Guide ⇒ iOS ⇒ Integration ⇒ Initialize the SDK</a>
+ */
++ (void)initializeSdkWithCompletionHandler:(nullable ALSdkInitializationCompletionHandler)completionHandler;
+
+/**
+ * Gets a shared instance of AppLovin SDK.
+ *
+ * @warning Make sure your SDK key is set in the application’s `Info.plist` under the property `AppLovinSdkKey`.
+ *
+ * @return The shared instance of AppLovin’s SDK, or `nil` if SDK key is not set in the application’s Info.plist.
+ */
++ (nullable ALSdk *)shared;
+
+/**
+ * Gets a shared instance of AppLovin SDK.
+ *
+ * @warning Make sure your SDK key is set in the application’s `Info.plist` under the property `AppLovinSdkKey`.
+ *
+ * @param settings An SDK settings object.
+ *
+ * @return The shared instance of AppLovin’s SDK, or nil if SDK key is not set in the application’s Info.plist.
+ */
++ (nullable ALSdk *)sharedWithSettings:(ALSdkSettings *)settings;
+
+/**
+ * Gets an instance of AppLovin SDK by using an SDK key.
+ *
+ * @param key SDK key to use for the instance of the AppLovin SDK.
+ *
+ * @return An instance of AppLovinSDK, or `nil` if the SDK key is not set.
+ */
+ // [PLP] what does it mean for the key to be "not set"? Does this mean "if you passed in a null or empty string for the key" or something more subtle?
```

Comment:
Hmmmm pubs should ALWAYs set an SDK key if they use this method, otherwise it's an error case, for which we will return `nil` (which is why we marked the return value as `nullable`).